### PR TITLE
Migrate remaining tests from `HudsonTestCase` to `JenkinsRule`

### PR DESCRIPTION
**Pull request by basil**
Previously: #5456. `HudsonTestCase` was deprecated in favor of `JenkinsRule` on June 12, 2013, yet 9 tests in this repository continue to use it. This PR migrates the remaining tests to `JenkinsRule`.

### Notes

- Commit c3c01c0335 (from December 2017) added a TODO in `RelativePathTest`. I completed this TODO by renaming the `name` property to `personName`.
- `GetEnvironmentOutsideBuildTest` was added in commit 83822830cf (from November 2011) and was described at that time as "unfortunately yet disfunctional [sic]". Rather than trying to get it to work I just deleted it.
- `RekeySecretAdminMonitorTest#testBasicWorkflow` had been flaky since it was first introduced in commit 4895eaafca (from January 2013). It was later ignored in commit f6662a8f6a (from May 2013). While I didn't find the root cause of the flakiness, I did find a workaround in test code that gets the test to pass reliably, so I enabled it again.

### CODE REVIEW

File: test/src/test/java/jenkins/security/RekeySecretAdminMonitorTest.java
Comment (daniel-beck) [inline test/src/test/java/jenkins/security/RekeySecretAdminMonitorTest.java:150]: Interesting. Is this Groovy related, or just `f:submit` in `message` views in general? Do we track this somewhere?

Also,
```suggestion
         * TODO There is a long-undiagnosed issue with the test harness not being compatible with
         * message.groovy's f.submit. Work around this by matching the behavior of a real browser
```

**Review by res0nance — APPROVED** [review]

File: test/src/test/java/jenkins/security/RekeySecretAdminMonitorTest.java
Comment (basil) [reply test/src/test/java/jenkins/security/RekeySecretAdminMonitorTest.java:150]: Not sure. I think commit f6662a8 comes closest to describing the problem.

**Review by StefanSpieker — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]
Ship it! Thanks @basil !

**Pull request merged**
